x-pack/plugins/lists/common/types.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * This makes any optional property the same as Required<T> would but also has the
+ * added benefit of keeping your undefined.
+ *
+ * For example:
+ * type A = RequiredKeepUndefined<{ a?: undefined; b: number }>;
+ *
+ * will yield a type of:
+ * type A = { a: undefined; b: number; }
+ *
+ */
+export type RequiredKeepUndefined<T> = { [K in keyof T]-?: [T[K]] } extends infer U
+  ? U extends Record<keyof U, [unknown]>
+    ? { [K in keyof U]: U[K][0] }
+    : never
+  : never;
+
+/**
+ * This is just a helper to cleanup nasty intersections and unions to make them
+ * readable from io.ts, it's an identity that strips away the uglyness of them.
+ */
+export type Identity<T> = {
+  [P in keyof T]: T[P];
+};
```

Comment:
NOTE: This identity will remove the hard to read intersection types from io-ts as it just iterates over keys and values and flattens them.

Before:
<img width="595" alt="Screen Shot 2020-05-04 at 9 06 39 AM" src="https://user-images.githubusercontent.com/1151048/80981016-98933480-8de6-11ea-9290-e86690b117ef.png">

After:
 <img width="714" alt="Screen Shot 2020-05-04 at 9 06 31 AM" src="https://user-images.githubusercontent.com/1151048/80981027-9c26bb80-8de6-11ea-952c-0d7f9044263a.png">